src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -217,6 +218,44 @@ private class SerializableLocalToolsManifest
             [JsonProperty(Required = Required.Always)]
             // The dictionary's key is the package id
             public Dictionary<string, SerializableLocalToolSinglePackage> tools { get; set; }
+
+            public SerializableLocalToolsManifest With(Dictionary<string, SerializableLocalToolSinglePackage> newTools)
+            {
+                return new SerializableLocalToolsManifest
+                {
+                    version = this.version,
```

Comment:
Not related to this change, but I would have loved for us to have used proper .NET naming conventions for these properties and annotated them for the JSON serialization, rather than using JavaScript naming convention inside C#.

Something we could fix for a future commit.

Reply:
We could do it, but we have to annotated every field like "Version" to have an explicit "json name" -- "version". I think since that class is created just for serialization, it is ok to have javascript name

Reply:
good point

Reply:
fixed